server/spike/adk/spike_02_caching.py

```
@@ -0,0 +1,225 @@
+                "prompt_tokens": prompt_tokens,
+                "cached_tokens": cached,
+                "output_tokens": output_tokens,
+            })
+            print(f"  USAGE: prompt={prompt_tokens}, cached={cached}, output={output_tokens}")
+        else:
+            usage_per_turn.append({"turn": i + 1, "prompt_tokens": 0, "cached_tokens": 0, "output_tokens": 0})
+            print("  USAGE: not available")
+
+    # ── Validation ────────────────────────────────────────────────────────────
+
+    print(f"\n── Results ──")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)